Fix DPS Sample documentation (#944)
Azure/azure-sdk-for-c: sdk/samples/iot/provisioning/README.md

## Introduction
This document explains samples for the Azure Embedded C SDK IoT Device Provisioning Client and how to use them.

> Samples are designed to highlight the function calls required to connect with the Device Provisioning Service (DPS). These calls illustrate the happy path of the [mqtt state machine](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/docs/iot/mqtt_state_machine.md). As a result, **these samples are NOT designed to be used as production-level code**. Production code needs to incorporate other elements, such as connection retries and more extensive error-handling, which these samples do not include. These samples also utilizes OpenSSL, which is **NOT recommended to use in production code on Windows or macOS**.

Reviewer: ```suggestion
Samples are designed to highlight the function calls required to connect with the Device Provisioning Service (DPS). These calls illustrate the happy path of the [mqtt state machine](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/docs/iot/mqtt_state_machine.md). As a result, **these samples are NOT designed to be used as production-level code**. Production code needs to incorporate other elements, such as connection retries and more extensive error-handling, which these samples do not include. These samples also utilize OpenSSL, which is **NOT recommended to use in production code on Windows or macOS**.
```

Author: what did you change?

Reviewer 2: utilizes -> utilize

	  ```
	 * For Windows systems, have [Microsoft Visual Studio](https://visualstudio.microsoft.com/downloads/) installed.
	 * For all systems, have the latest version of [CMake](https://cmake.org/download) installed.
* Have Microsoft [VCPKG](https://github.com/microsoft/vcpkg) package manager and [Eclipse Paho MQTT C client](https://www.eclipse.org/paho/) installed. Use the directions [here](https://github.com/Azure/azure-sdk-for-c#development-environment) to download VCPKG and install Paho MQTT.

## Getting Started
### Environment Variables
Samples use environment variables for a variety of purposes, including filepaths and connection parameters. Please keep in mind, every time a new terminal is opened, the environment variables will have to be reset. Setting a variable will take the following form:

Linux:
```bash
export ENV_VARIABLE_NAME=VALUE
```

Windows:
```cmd
set ENV_VARIABLE_NAME=VALUE
```

Set the following environment variables:
  * `VCPKG_DEFAULT_TRIPLET` and `VCPKG_ROOT`: Refer to these [directions](https://github.com/Azure/azure-sdk-for-c#development-environment).

>   * `AZ_IOT_DEVICE_X509_TRUST_PEM_FILE`: **Only for Windows or if required by OS.** Download [BaltimoreCyberTrustRoot.crt.pem](https://cacerts.digicert.com/BaltimoreCyberTrustRoot.crt.pem) to `/azure-sdk-for-c/sdk/samples/iot/`. Copy the entire filepath to this downloaded .pem file, e.g. `C:/azure-sdk-for-c/sdk/samples/iot/BaltimoreCyberTrustRoot.crt.pem`.

Reviewer: nit:
```suggestion
  * `AZ_IOT_DEVICE_X509_TRUST_PEM_FILE`: **Only for Windows or if required by OS.** Download [BaltimoreCyberTrustRoot.crt.pem](https://cacerts.digicert.com/BaltimoreCyberTrustRoot.crt.pem) to `/azure-sdk-for-c/sdk/samples/iot/`. Copy the full filepath to this downloaded .pem file, e.g. `C:/azure-sdk-for-c/sdk/samples/iot/BaltimoreCyberTrustRoot.crt.pem`.
```

* Have Microsoft [VCPKG](https://github.com/microsoft/vcpkg) package manager and [Eclipse Paho MQTT C client](https://www.eclipse.org/paho/) installed.  Use the directions [here](https://github.com/Azure/azure-sdk-for-c#development-environment) to download VCPKG and install Paho MQTT.

## Getting Started
### Environment Variables

> Samples use environment variables for a variety of purposes, including filepaths and connection parameters. Please keep in mind, every time a new terminal is opened, the environment variables will have to be reset.  Setting a variable will take the following form:

Reviewer 2: nit: We seem to have a bunch of double white space `  `. Find and remove them.
```suggestion
Samples use environment variables for a variety of purposes, including file paths and connection parameters. Please keep in mind, every time a new terminal is opened, the environment variables will have to be reset. Setting a variable will take the following form:
```

Author: I did some reading on this, and filepath as one word in this context is completely acceptable.  https://english.stackexchange.com/questions/428215/filepath-or-file-path#:~:text=A%20filepath%20is%20an%20entity%2C%20independent%20of%20any,not%20specifying%20what%20particular%20file%20we%27re%20talking%20about%29.#:~:text=A%20filepath%20is%20an%20entity%2C%20independent%20of%20any,not%20specifying%20what%20particular%20file%20we%27re%20talking%20about%29.#:~:text=A%20filepath%20is%20an%20entity%2C%20independent%20of%20any,not%20specifying%20what%20particular%20file%20we%27re%20talking%20about%29.

Linux:
```bash
export ENV_VARIABLE_NAME=VALUE
```

Windows:
```cmd

> set "ENV_VARIABLE_NAME=VALUE"

Reviewer 2: nit: I don't think this has to be in quotes (unless maybe there are white space characters).
```suggestion
set ENV_VARIABLE_NAME=VALUE
```

```

> Set the following environment  variables:

Reviewer 2: ```suggestion
Set the following environment variables:
```

>   * `VCPKG_DEFAULT_TRIPLET` and `VCPKG_ROOT`: Refer to  these [directions](https://github.com/Azure/azure-sdk-for-c#development-environment).

Reviewer 2: Same nit here/elsewhere. Will stop mentioning.
```suggestion
  * `VCPKG_DEFAULT_TRIPLET` and `VCPKG_ROOT`: Refer to these [directions](https://github.com/Azure/azure-sdk-for-c#development-environment).
```

Author: I will see what I can find, but I am not going to worry about this.  As you said, it always renders as a single space.

Reviewer 2: Sounds good. I don't know how this render in doxygen docs though and if the spaces are preserved there. Double check that and if it is an issue there, a Find/Replace of two (or more) spaces `  ` should help capture most of them.

Author: that's a good idea.  I used to always work in vim land so I forget about that.  thanks :)

  * `AZ_IOT_DEVICE_X509_TRUST_PEM_FILE`: **Only for Windows or if required by OS.** Download [BaltimoreCyberTrustRoot.crt.pem](https://cacerts.digicert.com/BaltimoreCyberTrustRoot.crt.pem) to `/azure-sdk-for-c/sdk/samples/iot/`. Copy the entire filepath to this downloaded .pem file, e.g. `C:/azure-sdk-for-c/sdk/samples/iot/BaltimoreCyberTrustRoot.crt.pem`.

### Paho IoT Provisioning (Certificates)
**Executable:** paho_iot_provisioning_example

> This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c) uses x509 authentication to connect to Azure IoT Hub Device Provisioning Service (DPS).  To easily run this sample, we have provided a script to generate a self-signed device certification used for device authentication.    **This script is intended for sample use only and not to be used in production code**.

Reviewer 2: Remove extra white space, it doesn't render in markdown on GitHub anyway.

```suggestion
This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c) uses x509 authentication to connect to Azure IoT Hub Device Provisioning Service (DPS).  To easily run this sample, we have provided a script to generate a self-signed device certification used for device authentication. **This script is intended for sample use only and not to be used in production code**.
```

    ```cmd
    generate_certificate.cmd
    ```
2. In your Azure DPS, add a new individual device enrollment using the recently generated `device_ec_cert.pem` file.  See [here](https://docs.microsoft.com/en-us/azure/iot-dps/quick-create-simulated-device-x509#create-a-device-enrollment-entry-in-the-portal) for further instruction.  After creation, the Registration ID of your device should appear as `paho-sample-device1` in the Individual Enrollments tab.
3. Set the following environment variables:
*  `AZ_IOT_DEVICE_X509_CERT_PEM_FILE`: Copy the path of the generated .pem file noted at the bottom of the generate_certificate output.
*  `AZ_IOT_REGISTRATION_ID`: This should be `paho-sample-device1`.
*  `AZ_IOT_ID_SCOPE`: Copy this value from the Overview tab in your Azure DPS.

### Paho IoT Provisioning (SAS)

> **Executable:** paho_iot_provisioning_sas_example

Reviewer: Shouldn't all of these be called `_sample` instead of `_example` since that's the root directory for these?

Author: yes i agree.  However, since they aren't all named that yet, I thought it best to at least be consistent and then update them all and the documentation at the same time (in case it gets forgotten about!)

This [sample](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/provisioning/src/paho_iot_provisioning_sas_example.c) uses SAS symmetric key authentication to connect to Azure IoT Hub Device Provisioning Service (DPS).

> 1. In your Azure DPS, add a new individual device enrollment using SAS. See [here](https://docs.microsoft.com/en-us/azure/iot-dps/quick-create-simulated-device-symm-key#create-a-device-enrollment-entry-in-the-portal) for further instruction. After creation, the Registration ID of your device will appear in the Individual Enrollments tab.

Reviewer 2: nit: Add back-ticks.
```suggestion
1. In your Azure DPS, add a new individual device enrollment using SAS. See [here](https://docs.microsoft.com/en-us/azure/iot-dps/quick-create-simulated-device-symm-key#create-a-device-enrollment-entry-in-the-portal) for further instructions. After creation, the `Registration ID` of your device will appear in the `Individual Enrollments` tab.
```

Author: I disagree.  Those are titles of items you find in the browser, not something you would type into or read off the command line, and hence would also be distracting here.

Reviewer 2: Fair enough :)

```suggestion
1. In your Azure DPS, add a new individual device enrollment using SAS. See [here](https://docs.microsoft.com/en-us/azure/iot-dps/quick-create-simulated-device-symm-key#create-a-device-enrollment-entry-in-the-portal) for further instructions. After creation, the Registration ID of your device will appear in the Individual Enrollments tab.
```
